src/main/java/com/sap/sgs/phosphor/fosstars/data/github/UnpatchedVulnerabilities.java

```diff
@@ -44,6 +60,71 @@ protected Feature supportedFeature() {
   @Override
   protected Value fetchValueFor(GitHubProject project) {
     logger.info("Figuring out if the project has any unpatched vulnerability ...");
-    return UNPATCHED_VULNERABILITIES.value(unpatchedVulnerabilities.getFor(project.url()));
+
+    Vulnerabilities vulnerabilities = new Vulnerabilities();
+    vulnerabilities.add(knownUnpatchedVulnerabilities.getFor(project.url()));
+    vulnerabilities.add(vulnerabilitiesFromNvdFor(project));
+
+    return UNPATCHED_VULNERABILITIES.value(vulnerabilities);
   }
+
+  /**
+   * Looks for unpatched vulnerabilities in NVD for a specified project.
+   *
+   * @param project The project.
+   * @return Unpatched vulnerabilities.
+   */
+  Vulnerabilities vulnerabilitiesFromNvdFor(GitHubProject project) {
+    Vulnerabilities vulnerabilities = new Vulnerabilities();
+    for (NvdEntry entry : nvd.search(VendorDataMatcher.with(project))) {
+      if (isUnpatched(entry)) {
+        vulnerabilities.add(
+            Vulnerability.Builder.from(entry)
+                .set(UNPATCHED)
+                .noFixDate()
+                .make());
+      }
+    }
+    return vulnerabilities;
+  }
+
+  /**
+   * Checks if an entry from NVD looks like an unpatched vulnerability
+   * by analyzing configurations.
+   *
+   * @param entry The entry from NVD.
+   * @return True if the entry looks like an unpatched vulnerability, false otherwise.
+   */
+  private static boolean isUnpatched(NvdEntry entry) {
+    if (entry.getConfigurations() == null || entry.getConfigurations().getNodes() == null) {
+      return false;
+    }
+
+    for (Node node : entry.getConfigurations().getNodes()) {
+      if (node == null || node.getCpeMatch() == null) {
+        continue;
+      }
+
+      for (CpeMatch cpeMatch : node.getCpeMatch()) {
```

Comment:
shouldn't the method name be node.getCpeMatches* ?

Reply:
This class and methods were generated from the JSON schema. We try to to touch them much although this pull request adds a couple of new methods to the class. If we decide to rename methods, then we should do it for the all generated classes. This is out of scope here.

Reply:
I didn't know JSON schema was that dumb!

Reply:
Sure. You can tell the NVD team about it.